### ISSUE TITLE
Add input fields for Maven coordinates in UI

### ISSUE THREAD

**Issue by geziefer** [eclipse-ee4j/starter#274]
The current version of the starter generates default Maven coordinates:
```
	<groupId>org.eclipse</groupId>
	<artifactId>jakartaee-hello-world</artifactId>
	<version>0.1-SNAPSHOT</version>

```

As it is common in other starters (e.g. SpringBoot, MicroProfile, Quarkus, OpenLiberty, ...) we should allow for editing those values in UI before project creation.

**Comment by geziefer** [comment]
I will reduce the Maven coordinates to groupId and artifactId, since version is not likely to change and other starters, like Microprofile or OpenLiberty does not ask for it as well. 
The others are not only importand as maven coordinates, but also for generating the Java packages.

**Comment by m-reza-rahman** [comment]
Great. I suspect the fewer the better.

**Comment by geziefer** [comment]
The feature was implemented as follows:
groupid and artifactid were added in ui as text input fields. They are both in one line to save space.
There is validation involved, making sure that they are not empty and contain only valid values. Group was validated for Java package naming and Artifact for Java jar naming (somewhat more strict than really possible, but I'd say reasonable).
The archetype was left untouched, but fed with the new parameters groupId, artifactId and package.
For generation, the mechanism as it was was also kept by creating the java package according to the groupId plus the fixed jakarta.hello for the sample code.
The name in pom and for the jar file was kept - here we could discuss if it should be named after artifactId, but in that case we have to adapt the archetype.
Concerning caching, it seemed to be reasonable to only cache when the default values for groupId and artifactId were used as otherwise it is highly unlikely that we profit from it and would only fill up the cache unnecessarily. 
Additionally, a few log statements were deleted which seem more for debugging while implementing ui selection.

**Comment by m-reza-rahman** [comment]
To fully implement this, this should really be closed first: https://github.com/eclipse-ee4j/starter/issues/189.

**Comment by m-reza-rahman** [comment]
These are related to this issue and can probably be closed together: https://github.com/eclipse-ee4j/starter/issues/151, https://github.com/eclipse-ee4j/starter/issues/150, https://github.com/eclipse-ee4j/starter/issues/149.

**Comment by geziefer** [comment]
Ok, I will have a look to the whole bundle when I find time for it, could take a bit currently.

**Comment by m-reza-rahman** [comment]
Will do. Let me know if you have questions.

**Comment by geziefer** [comment]
As discussed with Reza, we have to solve it completely, including the archetype.
So, the PR of this ticket was withdrawn and I will have a look on the complete package at the end of this year.

**Comment by geziefer** [comment]
This issue was addressed by PR #291 and PR #292 
It was done in 2 PRs as it affects different things and also includes field validation, #149 and #150

**Comment by m-reza-rahman** [comment]
Fixed
